File: 2022-2023/sourcecode/lab/lab10/ShoppyJava/app/src/main/java/pl/udu/uwr/pum/shoppyjava/fragments/list/ItemViewHolder.java
package pl.udu.uwr.pum.shoppyjava.fragments.list;

import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import pl.udu.uwr.pum.shoppyjava.data.Item;
import pl.udu.uwr.pum.shoppyjava.databinding.ItemRecyclerviewBinding;

public class ItemViewHolder extends RecyclerView.ViewHolder {

    private final ItemRecyclerviewBinding binding;

    public ItemViewHolder(@NonNull ItemRecyclerviewBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bind(Item item){
        binding.nameTextViewRV.setText(item.getName());
        binding.quantityTextViewRV.setText(String.valueOf(item.getQuantity()));
    }
}

File: 2022-2023/sourcecode/lab/lab10/ShoppyKotlin/app/src/main/java/pl/udu/uwr/pum/shoppykotlin/fragments/list/ItemViewHolder.kt
package pl.udu.uwr.pum.shoppykotlin.fragments.list

import androidx.recyclerview.widget.RecyclerView
import pl.udu.uwr.pum.shoppykotlin.data.Item
import pl.udu.uwr.pum.shoppykotlin.databinding.ItemRecyclerviewBinding

class ItemViewHolder(private val binding: ItemRecyclerviewBinding) : RecyclerView.ViewHolder(binding.root) {

    fun bind(item: Item) {
        binding.nameTextViewRV.text = item.name
        binding.quantityTextViewRV.text = item.quantity.toString()
    }
}